### PR TITLE
Final Version

### PR DESCRIPTION
**Pull request by Dineshg49**
Includes Procedure, Pretest, Postest, and Final Code

### CODE REVIEW

File: experiment/simulation/js/main.js [hunk 1]
```diff
@@ -1 +1,377 @@
...
+			[startx - 300, 4 * height + 40],
+			[startx - 250, 4 * height - 40],
+			[endx + 300, 4 * height - 40],
+			[endx + 250 , 4 * height + 40],
+		];
+
+		a = -1 * m1 * m2 * m3;
+		b = m1 * m2 * k3 + m1 * m3 * (k2 + k3) + m2 * m3 * (k1 + k2);
+		c = k2 * k2 * m3 - m1 * k3 * (k3 + k2) - m2 * (k1 + k2) * k3 - m3 * (k1+ k2) * (k2 + k3) + m1 * k3 * k3;
+		d = (k1 + k2) * (k2 + k3) * (k3) - k2 * k2 * k3 - k3 * k3 * (k1 + k2);
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js]: Please use descriptive names for all these variables. a,b,c,d,kn,mn do not convey any information when I read the code

File: experiment/simulation/js/main.js [hunk 2]
```diff
@@ -1 +1,377 @@
...
+
+		lambda = CubicSolve(a, b, c, d);
+		lambda.sort();
+
+		if(mode == 1)
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js:34]: Please use a descriptive enum for the mode types instead of bare numbers

**Review by raj-vlabs — CHANGES_REQUESTED** [review]
1. Multiple variable names carry no meaning. They need to be changed
2. Please use an enum for mode values. enums are not natively supported on JS, however, there are good alternatives available.
3. In main.js, a lot of code in restart function looks similar to the main code. Is it possible to restructure the code and have one function called with separate set of arguments. If the number of arguments is too high, they can be encapsulated in a map.

File: experiment/references.md [hunk 3]
```diff
@@ -1 +1,3 @@
-### Link your references in here
\ No newline at end of file
+- [https://ocw.mit.edu/NR/rdonlyres/Aeronautics-and-Astronautics/16-20Structural-%20MechanicsFall2002/7E90098F-C973-4B10-9C91-94730623C3B9/0/unit22.pdf](http://ocw.mit.edu/NR/rdonlyres/Aeronautics-and-Astronautics/16-20Structural-%20MechanicsFall2002/7E90098F-C973-4B10-9C91-94730623C3B9/0/unit22.pdf)
+- [https://www.aerostudents.com/files/vibrations/multipleDegreeOfFreedomSystems.pdf](http://www.aerostudents.com/files/vibrations/multipleDegreeOfFreedomSystems.pdf)
+- "Earthquake Resistant Design of Structures" by Pankaj Agarwal, Manish Shrikande
```
Comment (raj-vlabs) [inline experiment/references.md]: Please use https links only. Also do not use link in the description of the reference. Please use a descriptive phrase/sentence.

File: experiment/simulation/js/main.js (hunk 2 above)
Comment (Dineshg49) [reply experiment/simulation/js/main.js:34]: The value of the mode is self-descriptive as it shows the different modes of vibration of the system (i.e first, second and third modes of vibration). So using enum would be redundant.

File: experiment/simulation/css/main.css [hunk 4]
```diff
@@ -1 +1,24 @@
-/* You CSS goes in here */
\ No newline at end of file
+/* You CSS goes in here */
```
Comment (raj-vlabs) [inline experiment/simulation/css/main.css]: Please remove this comment.

File: experiment/simulation/index.html [hunk 5]
```diff
@@ -1,13 +1,99 @@
...
+	<head>
+		<!-- Add CSS at the head of HTML file -->
+		<meta charset="utf-8">
+		<meta name="viewport" content="width=device-width, initial-scale=1">
+		<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.1/css/bulma.min.css">
+		<link rel="stylesheet" href="./css/main.css">
+	</head>
+
+	<body class="page">
 		<!-- Your code goes here-->
```
Comment (raj-vlabs) [inline experiment/simulation/index.html]: Please remove this comment.

File: experiment/simulation/index.html [hunk 6]
```diff
@@ -1,13 +1,99 @@
...
+						<canvas class="main" id="main"></canvas>
+					</div>
+				</div>
+			</div>
+		</div>
 
-	<!-- Add JS at the bottom of HTML file --> 
-        <script src="./js/main.js"></script>
-</body>
+		<!-- Add JS at the bottom of HTML file -->
```
Comment (raj-vlabs) [inline experiment/simulation/index.html]: Please remove this comment.

**Review by raj-vlabs — CHANGES_REQUESTED** [review]
Please remove the irrelevant comments.

**Review by raj-vlabs — APPROVED** [review]

**Pull request merged**